### DISCUSSION TITLE
Please make FlexLayout really usable!!!

### DISCUSSION THREAD

**Discussion by niltor** [dotnet/maui#15311] [Q&A]
## confused
Is there a layout system in MAUI?
In the 21st century, can we only use complex and cumbersome GridLayout layouts? And stackLayout, which has almost no practical effect?

Why can't we use a layout system similar to float or flex in css?

In recent years, I tried to use maui several times to start new projects. When flexlayout appeared, I thought I could use my familiar flex knowledge to layout, but it was completely impossible!

## The problem
When flex is nested with other layouts, it cannot be displayed normally. When flex and flex itself are nested, they cannot be displayed normally. Why?
There are so many counter-intuitive designs that make us generate too much negative feedback when writing xaml.

For example, aligning an element to the right, or occupying 30% of the width, this is a very simple requirement, but it is too abstract to implement in xaml!

In today's wave of AI development, why do we developers waste our time and energy on the most basic layout?

## We have to think about what is really beneficial to developers?
Free layout and typesetting without being limited by the type of layout used, just like div.

This Includes:

- The width and height can be freely adjusted using percentages
- Unified layout system instead of various layout controls
- Free and easy to modify the control style

Or in a simple sentence, so that all developers who understand css can use a simple tag, such as <div> or <layout>, to make a free layout without learning a specific and uncommon layout design！

**Comment by Xyncgas** [comment]
html is better than XAML, they won't admit it and they won't provide a way for you to paint UI with position and dimension and UI event handler and styling in code, you have to use xaml OR ELSE

>aligning an element to the right, or occupying 30% of the width, this is a very simple requirement, but it is too abstract to implement in xaml!

you're right
if it's text, just do ```text-align:right;``` in css
if it's div, just do ```display:inline-block;width:30%;margin-left:70%;``` and put that inside another div that has its width equal to the row you are putting it in

>Free layout and typesetting without being limited by the type of layout used, just like div.

I've advocated the use of position and dimension and styling and UI events for drawing custom layouts

It's basically everything needed in 2D UI development : Of course, sometimes you want to display image or vector graph or openGL content so you kinda have different components that are specialized for those purposes that might be without attributes like dimensions or positions instead taking out the entire space provided by the containing element

forget about UI diff-ing algorithm forget about providing all kinds of different components, new frameworks emerges from these basic ingredients will take care of that, people can write framework for MAUI like people did React.js for html, if things doesn't work in the framework they are using they can always do it themselves

**Comment by hartez** [comment]
If you're having problems using FlexLayout, please submit an issue.

**Reply by niltor** [reply]
There are enough existing issues😅

Not for patching,
Instead, we need to understand the official views and plans on Flexlayout, what it will look like in its final usable state, so that we can evaluate whether to give up directly or to look forward to it.

**Reply by hartez** [reply]
The plan is for FlexLayout to provide all of the functionality it provided in Xamarin.Forms.
